### ISSUE TITLE
Files/SideEffectsSniff does not respect @codingStandardsIgnoreLine directive

### ISSUE THREAD

**Issue by Hoten** [squizlabs/PHP_CodeSniffer#1698]
Trying to hide one error that I'd wish to ignore, but the ignoring directives don't work as expected.

```php
<?php

namespace Reqy;

/**
 * Class ReqyErrorLevel
 * @package Reqy
 */
class ReqyErrorLevel
{
    /** @var ReqyErrorLevel */
    public static $ERROR;

    /** @var ReqyErrorLevel */
    public static $WARNING;

    public static function initClass()
    {
        self::$ERROR = new ReqyErrorLevel('ERROR');
        self::$WARNING = new ReqyErrorLevel('WARNING');
    }

    /** @var string */
    protected $level;

    /**
     * ReqyErrorLevel constructor.
     * @param string $level
     */
    public function __construct($level)
    {
        $this->level = $level;
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->level;
    }
}

// @codingStandardsIgnoreLine
ReqyErrorLevel::initClass();

```

Running the linter `./vendor/bin/phpcs --standard=PSR2 src/`:
```
--------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
--------------------------------------------------------------------------------------------------------------------------------------------------------------
 1 | WARNING | A file should declare new symbols (classes, functions, constants, etc.) and cause no other side effects, or it should execute logic with side
   |         | effects, but should not do both. The first symbol is defined on line 9 and the first side effect is on line 45.
--------------------------------------------------------------------------------------------------------------------------------------------------------------

```

Same thing if I wrap the entire file with the start/end directives.

**Comment by gsherwood** [comment]
The way you expect that feature to work, and what it actually does, is not the same. Take a look at issue #1258 and my first comment on there.

I'm going to close this because the other issue is an exact duplicate, and it would be better to keep all the comments together. Please feel free to comment over there even though it is closed.
